app/controllers/tournaments_controller.rb

```diff
@@ -20,6 +42,18 @@ def scores
   end
 
   def success
+    unless Tournament.exists?(params[:id])
```

Comment:
To perform less queries, you can select the tournament with the results:
```ruby
tournament = Tournament.find_by(id: params[:id]).includes(:results) # this prefetches the results when querying
if tournament.blank?
  render ...
  return
end
results = tournament.results
```

Reply:
It works without the _includes_. When I try to add it, I get _undefined method `includes`_

Reply:
@MariaGunkoCX , try this:
```
Tournament.find_by(id: params[:id], include: :results)
```

Reply:
Also, the opposite order might work:
```
Tournament.includes(:results).find_by(....)
```

Reply:
Thanks Or :) yeah I messed up the order

Reply:
**Tournament.find_by(id: params[:id], include: :results)**
It fails with _Mysql2::Error: Unknown column 'tournaments.include' in 'where clause'_ 👎 
**Tournament.includes(:results).find_by(....)** 
It works 👍 

Thank you both :)